namespace/joiner.go

```diff
@@ -0,0 +1,390 @@
+package namespace
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+
+	"github.com/longhorn/go-common-libs/types"
+	"github.com/longhorn/go-common-libs/utils"
+)
+
+type Joiners []*Joiner
```

Comment:
When will there be multiple joiners?

Reply:
To reach the final namespace, [multiple ns fds](https://github.com/longhorn/go-common-libs/blob/a0a85547aa56a5927ea1739af916cf10fbb154a2/namespace/joiner.go#L181-L184) will need to be joined. Each joiner represents a namespace. ref: https://github.com/longhorn/go-common-libs/blob/a0a85547aa56a5927ea1739af916cf10fbb154a2/namespace/joiner.go#L253-L257.